test/feature_discovery_test.dart

```diff
@@ -4,86 +4,78 @@ import 'package:flutter_test/flutter_test.dart';
-        FeatureDiscovery.discoverFeatures(context, steps);
-        await tester.pumpAndSettle();
-        // First overlay should appear
-        expect(find.text(texts[0]), findsOneWidget);
-        // Test with a tap on the target
-        await tester.tap(finder.first);
-        await tester.pumpAndSettle();
-        // First overlay should have disappeared, and second appeared
-        expect(find.text(texts[0]), findsNothing);
-        expect(find.text(texts[1]), findsOneWidget);
-        // Test with [completeCurrentStep]
-        FeatureDiscovery.completeCurrentStep(context);
-        await tester.pumpAndSettle();
-        expect(find.text(texts[1]), findsNothing);
-        expect(find.text(texts[2]), findsOneWidget);
-        // Dismiss all
-        FeatureDiscovery.dismiss(context);
-        await tester.pumpAndSettle();
-        // No overlay should remain
-        texts.forEach((t) => expect(find.text(t), findsNothing));
-      }
-    );
+    testWidgets("Displaying two steps and dismissing before the third",
+        (WidgetTester tester) async {
+      await tester.pumpWidget(const TestWidget(featureIds: steps));
+      final Finder finder = find.byType(TestIcon);
+      expect(finder, findsNWidgets(steps.length));
+      final BuildContext context = tester.firstState(finder).context;
+      // Should be no overlays before calling discoverFeatures
+      texts.forEach((t) => expect(find.text(t), findsNothing));
+      FeatureDiscovery.discoverFeatures(context, steps);
+      await tester.pumpAndSettle();
+      // First overlay should appear
+      expect(find.text(texts[0]), findsOneWidget);
+      // Test with a tap on the target
+      await tester.tap(finder.first);
+      await tester.pumpAndSettle();
+      // First overlay should have disappeared, and second appeared
+      expect(find.text(texts[0]), findsNothing);
+      expect(find.text(texts[1]), findsOneWidget);
+      // Test with [completeCurrentStep]
+      FeatureDiscovery.completeCurrentStep(context);
+      await tester.pumpAndSettle();
+      expect(find.text(texts[1]), findsNothing);
+      expect(find.text(texts[2]), findsOneWidget);
+      // Dismiss all
+      FeatureDiscovery.dismiss(context);
+      await tester.pumpAndSettle();
+      // No overlay should remain
+      texts.forEach((t) => expect(find.text(t), findsNothing));
+    });
   });
 
   group("Non-existent featureIds", () {
-    const List<String> featureIds = [
-      "featA",
-      "featB",
-      "featC"
-    ];
+    const List<String> featureIds = ["featA", "featB", "featC"];
     final List<String> texts = textsToMatch(featureIds);
     testWidgets(
-      "Calling [discoverFeatures] with two ids that aren't associated with an overlay",
-      (WidgetTester tester) async {
-        await tester.pumpWidget(TestWidget(
+        "Calling [discoverFeatures] with two ids that aren't associated with an overlay",
+        (WidgetTester tester) async {
```

Comment:
Is this really a better syntax?

Reply:
I simply ran `flutter dartfmt`, which is what is required by the Pub checking system. If formatting is off, the plugin will lose some points.

Reply:
You're right, and this gave me the idea to add a formatting check to the repo test workflow. See axel-op#25.